File: src/main/java/net/teamhollow/direbats/client/DirebatsClient.java
package net.teamhollow.direbats.client;

import com.google.common.reflect.Reflection;
import net.fabricmc.api.ClientModInitializer;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.fabricmc.fabric.api.client.rendering.v1.EntityRendererRegistry;
import net.teamhollow.direbats.Direbats;
import net.teamhollow.direbats.client.model.DirebatsEntityModelLayers;
import net.teamhollow.direbats.client.render.entity.DirebatEntityRenderer;
import net.teamhollow.direbats.client.render.entity.DirebatFangArrowEntityRenderer;
import net.teamhollow.direbats.entity.DirebatsEntities;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

@Environment(EnvType.CLIENT)
public class DirebatsClient implements ClientModInitializer {
    public static final Logger LOGGER = LogManager.getLogger("%s-client".formatted(Direbats.MOD_ID));

    @SuppressWarnings("UnstableApiUsage")
    @Override
    public void onInitializeClient() {
        LOGGER.info("Initializing {}-client", Direbats.MOD_NAME);

        Reflection.initialize(DirebatsEntityModelLayers.class);

        EntityRendererRegistry.register(DirebatsEntities.DIREBAT, DirebatEntityRenderer::new);
        EntityRendererRegistry.register(DirebatsEntities.DIREBAT_FANG_ARROW, DirebatFangArrowEntityRenderer::new);

        LOGGER.info("Initialized {}-client", Direbats.MOD_NAME);
    }
}

File: src/main/kotlin/dev/andante/direbats/client/DirebatsClient.kt
package dev.andante.direbats.client

import dev.andante.direbats.client.render.entity.DirebatEntityRenderer
import dev.andante.direbats.client.render.entity.DirebatFangArrowEntityRenderer
import dev.andante.direbats.client.render.entity.model.DirebatsEntityModelLayers
import dev.andante.direbats.entity.DirebatsEntityTypes
import net.fabricmc.api.ClientModInitializer
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.fabricmc.fabric.api.client.rendering.v1.EntityRendererRegistry

@Environment(EnvType.CLIENT)
object DirebatsClient : ClientModInitializer {
    override fun onInitializeClient() {
        registerEntityRenderers()
    }

    private fun registerEntityRenderers() {
        DirebatsEntityModelLayers

        EntityRendererRegistry.register(DirebatsEntityTypes.DIREBAT, ::DirebatEntityRenderer)
        EntityRendererRegistry.register(DirebatsEntityTypes.DIREBAT_FANG_ARROW, ::DirebatFangArrowEntityRenderer)
    }
}
